flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java

```diff
@@ -61,6 +62,84 @@
 	private static final String FIRST_PART = "That is the first part";
 	private static final String SECOND_PART = "That is the second part";
 
+	@Test
+	public void testNullValueWithoutTrailingDelimiter() throws Exception {
```

Comment:
We should test the individual parsers by extending `ParserTestBase` with tests that check for correct empty field behavior (EMPTY_COLUMN error state) with and without a following delimiter.

IMO, the null field behavior of `RowCsvInputFormat` is sufficiently tested by `RowCsvInputFormatTest.testEmptyFields()` such that this test does not need to be extended.

Reply:
+1

Reply:
About `RowCsvInputFormatTest.testEmptyFields()`do not check the tailing empty field, And in  `RowCsvInputFormatTest.testTailingEmptyFields`, test tailing empty field for STRING type. I think we shoud add 
DOUBLE/FLOAT/BIGINT/BIGDEC/SQL_DATE/SQL_TIME/SQL_TIMESTAMP test case as well.
What do you think?